galsim/chromatic.py

```
@@ -421,6 +421,97 @@ def drawImage(self, bandpass, image=None, integrator='trapezoidal', **kwargs):
+            if integrator == 'trapezoidal':
+                rule = np.trapz
+            elif integrator == 'midpoint':
+                rule = galsim.integ.midpt
+            else:
+                raise TypeError("Unrecognized integration rule: %s"%integrator)
+            if len(wave_list) > 0:
+                integrator = galsim.integ.SampleIntegrator(rule)
+            else:
+                integrator = galsim.integ.ContinuousIntegrator(rule)
+        if not isinstance(integrator, galsim.integ.SampleIntegrator) and \
+                not isinstance(integrator, galsim.integ.ContinuousIntegrator):
+            raise TypeError("Invalid type passed in for integrator!")
+
+        # merge self.wave_list into bandpass.wave_list if using a sampling integrator
+        if isinstance(integrator, galsim.integ.SampleIntegrator):
+            bandpass = galsim.Bandpass(galsim.LookupTable(wave_list, bandpass(wave_list),
+                                                          interpolant='linear'),
+                                       wave_type='nm')
+
+        add_to_image = kwargs.pop('add_to_image', False)
+        image_int = integrator(self.evaluateAtWavelength, bandpass, image, kwargs, doK=True)
+
+        # For performance profiling, store the number of evaluations used for the last integration
+        # performed.  Note that this might not be very useful for ChromaticSum instances, which are
+        # drawn one profile at a time, and hence _last_n_eval will only represent the final
+        # component drawn.
+        self._last_n_eval = integrator.last_n_eval
+
+        # Apply integral to the initial image appropriately.
+        # Note: Don't do image = integral and return that for add_to_image==False.
+        #       Remember that python doesn't actually do assignments, so this won't update the
+        #       original image if the user provided one.  The following procedure does work.
+        if add_to_image:
+            image += image_int
```

There's a bug here too, which is related to `gsobject.drawKImage()` not having a `setup_only` kwarg, so I'm implementing that.  One question though: should we raise an exception in `gsobject.drawKImage()` if both `add_to_image` and `recenter` are True?  Seems like it might be surprising behavior to do both.